src/clients/tenant-client.js

```diff
@@ -70,14 +70,20 @@ class TenantClient extends BaseEvernodeClient {
         const hostAcc = await this.getLeaseHost(hostAddress);
         let selectedOfferIndex = options.leaseOfferIndex;
 
-        // Attempt to get first available offer, if offer is not specified in options.
-        if (!selectedOfferIndex) {
-            const uritOffers = await EvernodeHelpers.getLeaseOffers(hostAcc);
-            selectedOfferIndex = uritOffers && uritOffers[0] && uritOffers[0].index;
+        let buyUriOffer = null;
+        const uritOffers = await EvernodeHelpers.getLeaseOffers(hostAcc);
 
-            if (!selectedOfferIndex)
-                throw { reason: ErrorReasons.NO_OFFER, error: "No offers available." };
+        if (!selectedOfferIndex) {
+            // Attempt to get first available offer, if offer is not specified in options.
+            buyUriOffer = uritOffers && uritOffers[0];
         }
+        else {
+            // Attempt to get relevant available offer using selectedOfferIndex.
+            buyUriOffer = uritOffers && uritOffers.find(uriOffer => { uriOffer.index === selectedOfferIndex });
```

Comment:
I think we need to change `{ uriOffer.index === selectedOfferIndex }` to `uriOffer.index === selectedOfferIndex`, Otherwise If we add curly brackets it would be an arrow function so we need to return a boolean in that case.